File: app/src/main/java/com/lucas/omnia/models/Inspection.java
package com.lucas.omnia.models;

import android.content.res.Resources;

import com.lucas.omnia.R;
import com.lucas.omnia.activities.BlocksActivity;
import com.lucas.omnia.activities.BodiesActivity;
import com.lucas.omnia.activities.DeputiesActivity;
import com.lucas.omnia.activities.EventsActivity;
import com.lucas.omnia.activities.FrontsActivity;
import com.lucas.omnia.activities.PartiesActivity;
import com.lucas.omnia.activities.PollsActivity;
import com.lucas.omnia.activities.PropositionsActivity;

import java.util.ArrayList;

public class Inspection {

    private String name;
    private Class activityClass;

    public Inspection() {}

    private Inspection(String name, Class activityClass) {
        this.name = name;
        this.activityClass = activityClass;
    }

    public String getName() {
        return name;
    }

    public Class getActivityClass() {
        return activityClass;
    }

    public static ArrayList<Inspection> createInspectionList(Resources resources) {
        ArrayList<Inspection> inspectionList = new ArrayList<>();

        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_1),BlocksActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_9), BodiesActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_2),DeputiesActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_3),EventsActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_4),FrontsActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_6),PartiesActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_7),PropositionsActivity.class));
        inspectionList.add(new Inspection(resources.getString(R.string.inspection_item_8), PollsActivity.class));

        return inspectionList;
    }
}

File: app/src/main/java/com/lucas/omnia/models/Inspection.kt
package com.lucas.omnia.models

import android.content.res.Resources
import com.lucas.omnia.R
import com.lucas.omnia.activities.*
import java.util.*

data class Inspection(var name: String? = null, var activityClass: Class<*>? = null) {

    companion object {
        @JvmStatic
        fun createInspectionList(resources: Resources): ArrayList<Inspection> {
            val inspectionList = ArrayList<Inspection>()
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_1), BlocksActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_9), BodiesActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_2), DeputiesActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_3), EventsActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_4), FrontsActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_6), PartiesActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_7), PropositionsActivity::class.java))
            inspectionList.add(Inspection(resources.getString(R.string.inspection_item_8), PollsActivity::class.java))
            return inspectionList
        }
    }
}